engine/lib/views.php

```diff
@@ -813,25 +813,25 @@ function elgg_view_entity(\ElggEntity $entity, $vars = array(), $bypass = false,
 
 	$vars['entity'] = $entity;
 
-
-	// if this entity has a view defined, use it
-	$view = $entity->view;
-	if (is_string($view)) {
-		return elgg_view($view, $vars, $bypass, $debug);
-	}
-
 	$entity_type = $entity->getType();
-
-	$subtype = $entity->getSubtype();
-	if (empty($subtype)) {
-		$subtype = 'default';
+	$entity_subtype = $entity->getSubtype();
+	if (empty($entity_subtype)) {
+		$entity_subtype = 'default';
 	}
 
+	$entity_views = array(
+		$entity->view,
+		$entity->getVolatileData('view'),
```

Comment:
`item_view` sounds good. Could be used viewing annotations/other things. I could even see calling elgg_list_entities() with `'callback' => ''` to explicitly avoid the overhead of entity instantiation, but that's probably not something we want to go out of our way to support.